src/Build/Graph/GraphBuilder.cs

```diff
@@ -557,8 +589,15 @@ private List<ProjectInterpretation.ReferenceInfo> ParseReferences(ProjectGraphNo
                         referenceInfo.ReferenceConfiguration.ProjectFullPath
                         ));
                 }
-                
-                SubmitProjectForParsing(referenceInfo.ReferenceConfiguration);
+
+                if (!referenceInfo.ProjectReferenceItem.HasMetadata("SetPlatform"))
```

Comment:
I'm not sure we want this. If `SetPlatform` is already defined in a `ProjectReference`, we don't want to negotiate for that reference, but what about _its_ references?

This is an interesting case for a unit test, can you add this?

_A(plat: x64) - SetPlatform=x86 > B (plats: x64, AnyCPU) -> C (plats: x86, x64)_
A should be x64, B should be x86 (because the SetPlatform in A's `ProjectReference` item to B), C should be x86 (because B was x86 and C can be x86)

I suspect it will fail with the current logic. We'd get here when parsing A's references, see that B has SetPlatform set and fall into the else case. We'd then parse B's references with `enableDynamicPlatformResolution` being false and no negotiation would happen for C.

Reply:
Turns out this wasn't true because we're always pulling the current project's "EnableDynamicPlatformResolution" property and passing that down the chain.